### ISSUE TITLE
Distribution validation for zero variability 

### ISSUE THREAD

**Issue by rnugent3** [HydrologicEngineeringCenter/HEC-FDA#247]
If a user specifies a distribution with zero variability, i.e. a normal with standard deviation of zero or a triangular or uniform with min = max, then a deterministic distribution should be passed to the model, not normal/triangular/uniform.

**Comment by codymccoy** [comment]
When the user opens the editor back up to edit, do you want it to display the way that they had it, or should it display as a constant? I ask because if you want the latter we could just store it in the db as a constant when the user clicks save. Either way shouldn't be too difficult.

**Comment by HenryGeorgist** [comment]
i would offer a different solution all together. a validation rule on a coordinate that contains a normal distribution should error out a message if standard deviation is 0 that "the distribution is invalid please use deterministic distribution type if you want a standard deviation of zero"

Dont monkey around with changing things automatically for them - that is super dangerous and will end up with mad users. tell them what is ok and what is not ok, and let them choose.

**Comment by rnugent3** [comment]
@HenryGeorgist how would you propose users enter the EGM curves, then? 

Each curve has a coordinate with zero damage and zero standard deviation.

**Comment by rnugent3** [comment]
![image](https://user-images.githubusercontent.com/25163165/150610389-75d65229-5d41-48c0-b75a-547ef090b880.png)

**Comment by rnugent3** [comment]
Rating curves will have standard deviation of zero. How will we handle importing these?

It seems weird to tell users that their data was good in 1.4.2 but now their data is bad and we won't accept it, no to mention the problems this will cause in certification

**Comment by HenryGeorgist** [comment]
so we silently mutate the data instead? That seems like it is full of problems. this is why things are called "black boxes"

**Comment by rnugent3** [comment]
but we aren't really mutating the data. A normal distribution with standard deviation of zero is mathematically equivalent to a constant, or in our library, a deterministic distribution.

**Comment by HenryGeorgist** [comment]
alternatively in the normal distribution we could add a check in inversecdf if stdev=0 return mean

**Comment by rnugent3** [comment]
second option seems easier

**Comment by HenryGeorgist** [comment]
yeah, the taylor polinomial may be faster than inverse newton, but i believe the inverse newton is "closer" to reality. if we switched to taylor polinomial approximation we might also see a speed increase.

**Comment by HenryGeorgist** [comment]
ill address triangular, uniform, lognormal, and normal in statistics.
